src/tests/useInfiniteQuery.test.js

```diff
@@ -120,4 +128,89 @@ describe('useInfiniteQuery', () => {
       rendered.getByText('Page 1: 3'),
     ])
   })
+
+  it('should compute canFetchMore correctly using initialData', async () => {
+    function Page() {
+      const fetchCountRef = React.useRef(0)
+      const {
+        status,
+        data,
+        error,
+        isFetching,
+        isFetchingMore,
+        fetchMore,
+        canFetchMore,
+        refetch,
+      } = useInfiniteQuery(
+        'items',
+        (key, nextId = 0) => fetchItems(nextId, fetchCountRef.current++),
+        {
+          initialData: [initialItems(0), initialItems(1)],
+          getFetchMore: (lastGroup, allGroups) => lastGroup.nextId,
+        }
+      )
+
+      expect(canFetchMore).not.toBeUndefined()
```

Comment:
I don't think you can use expect inside a component.
I would rather render `canFetchMore` inside a dom element and check it after component is rendered.
(In this case `canFetchMore` should be stringified, so "undefined" can be rendered as string)

Reply:
Yep. Render that to a div or something and check that it exists with react testing library and this will be great!